### PR TITLE
Add organizationSpaceMemberships endpoints

### PR DESCRIPTION
**Pull request by burakukula**
As a part of the following task https://contentful.atlassian.net/browse/HEJO-1140 we would like to add /organization/<orgId>/space_memberships endpoints to the SDK.

### CODE REVIEW

File: lib/create-organization-api.js [hunk 1]
```diff
@@ -77,8 +79,53 @@ export default function createOrganizationApi ({
...
+      headers
+    })
+      .then(response => wrapOrganizationSpaceMembership(http, response.data, orgId), errorHandler)
+  }
+
+  /**
+   * Gets a collection of Organization Space Memberships
+   * @memberof ContentfulOrganizationAPI
+   * @param {object=} query - Object with search parameters. Check the <a href="https://www.contentful.com/developers/docs/javascript/tutorials/using-js-cda-sdk/#retrieving-entries-with-search-parameters">JS SDK tutorial</a> and the <a href="https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters">REST API reference</a> for more details.
+   * @return {Promise<OrganizationSpaceMembership.OrganizationSpaceMembershipCollection>} Promise for a collection of Organization Memberships
```
Comment (vbeleuta) [inline lib/create-organization-api.js]: small docs update: "collection of Organization Space Memberships"

File: test/integration/integration-tests.js [hunk 2]
```diff
@@ -207,6 +208,16 @@ test('Gets organization for tests which change and delete data', (t) => {
     })
 })
 
+test.only('Gets organization for tests which change and delete data', (t) => {
+  return client.getOrganizations()
+    .then((response) => {
+      const organization = response.items[0]
+      return Promise.all([
+        organizationSpaceMembershipTests(t, organization)
```
Comment (vbeleuta) [inline test/integration/integration-tests.js]: I do not think we need to get the organization two times, you can probably combine the organizationSpaceMembershipTests with the organizationMembershipTests. So move this line in the Promise.all([organizationMembershipTests(t, org), organizationSpaceMembershipTests(t, org)])

File: lib/create-organization-api.js (hunk 1 above)
Comment (vbeleuta) [inline lib/create-organization-api.js]: small docs update: "a collection of Organization Space Memberships"

File: test/integration/integration-tests.js (hunk 2 above)
Comment (burakukula) [reply test/integration/integration-tests.js]: @vbeleuta oh! yeah! totally. thx a lot for your feedback. 🙂

File: lib/create-organization-api.js (hunk 1 above)
Comment (burakukula) [reply lib/create-organization-api.js]: on my way! 🚐

File: test/integration/integration-tests.js (hunk 2 above)
Comment (burakukula) [reply test/integration/integration-tests.js]: Done 🙂

File: lib/entities/organization-space-membership.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * Organization space membership instances
+ * @namespace OrganizationSpaceMembership
+ */
+
+import cloneDeep from 'lodash/cloneDeep'
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import enhanceWithMethods from '../enhance-with-methods'
+
+/**
+ * @memberof OrganizationSpaceMembership
```
Comment (guilebarbosa) [inline lib/entities/organization-space-membership.js]: There isn't an OrganizationSpaceMembership entity in Contentful. What changes is the scoping.
`organizations/:orgId/space_memberships` will just return `SpaceMembership` objects, but across all spaces in the org.
So I would just remove these type definitions.

File: test/integration/integration-tests.js [hunk 4]
```diff
@@ -197,12 +198,13 @@ test('Gets v2 space for read only tests', (t) => {
     })
 })
 
-test('Gets organization for tests which change and delete data', (t) => {
+test.only('Gets organization for tests which change and delete data', (t) => {
```
Comment (guilebarbosa) [inline test/integration/integration-tests.js]: Oops. Should remove `.only`

File: test/integration/organization-space-membership-integration.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  t.test('Gets organizationSpaceMembership', (t) => {
+    t.plan(5)
+    return organization.getOrganizationSpaceMembership('527cW1W5JNlOlJq38lfr1k')
+      .then((response) => {
+        t.ok(response.sys, 'sys')
+        t.ok(response.sys.id, '527cW1W5JNlOlJq38lfr1k')
+        t.ok(response.sys.type, 'SpaceMembership')
+        t.ok(response.user, 'user')
+        t.ok(response.roles, 'roles')
```
Comment (guilebarbosa) [inline test/integration/organization-space-membership-integration.js]: `.ok()` only checks if a value exists. The second argument is just a message that get's logged in the console.
You should consider matching the values with `.equal`.

File: test/integration/organization-space-membership-integration.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+export default function organizationSpaceMembershipTests (t, organization) {
+  t.test('Gets organizationSpaceMemberships', (t) => {
+    t.plan(2)
+    return organization.getOrganizationSpaceMemberships()
+      .then((response) => {
+        t.ok(response.sys, 'sys')
+        t.ok(response.items, 'fields')
```
Comment (guilebarbosa) [inline test/integration/organization-space-membership-integration.js]: Should you check if `items` contain space memberships?

File: test/unit/mocks/entities.js [hunk 7]
```diff
@@ -183,6 +183,12 @@ const organizationMembershipMock = {
   })
 }
 
+const organizationSpaceMembershipMock = {
```
Comment (guilebarbosa) [inline test/unit/mocks/entities.js]: You can remove this and use `spaceMembershipMock` instead.
Also it seems that they are missing some properties like `admin` and `roles`. Don't know how important they are in the tests though.

File: lib/create-organization-api.js (hunk 1 above)
Comment (guilebarbosa) [reply lib/create-organization-api.js]: That would be misleading.
I would just declare `{Promise<SpaceMembershipCollection>}` and say "Collection of Space Memberships across all spaces in the organization

File: test/integration/integration-tests.js (hunk 4 above)
Comment (burakukula) [reply test/integration/integration-tests.js]: oopsyyyy. sorry

File: test/unit/mocks/entities.js (hunk 7 above)
Comment (burakukula) [reply test/unit/mocks/entities.js]: @guilebarbosa sure, on my way 🚐

File: lib/entities/organization-space-membership.js (hunk 3 above)
Comment (burakukula) [reply lib/entities/organization-space-membership.js]: yeah, you’re totally right.. I removed them. thx for feedback 🙂

File: test/integration/organization-space-membership-integration.js (hunk 6 above)
Comment (burakukula) [reply test/integration/organization-space-membership-integration.js]: @guilebarbosa I am checking it with the `getOrganizationSpaceMembership()` test. But maybe I would adjust it to check it more explicitly like: 
```
t.equal(response.sys.type, 'SpaceMembership', 'type')
```
WDYT?

Comment (guilebarbosa) [reply test/integration/organization-space-membership-integration.js]: Hmm, actually the type here would be `Array`.

Comment (burakukula) [reply test/integration/organization-space-membership-integration.js]: @guilebarbosa I was mentioning the test which I would add into the `getOrganizationSpaceMembership`. I am not sure if I should test it in here...

File: lib/entities/organization-space-membership.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+/**
```
Comment (guilebarbosa) [inline lib/entities/organization-space-membership.js]: I think the whole file can also go alway as the entity to be used is Space Membership.

File: lib/create-organization-api.js [hunk 9]
```diff
@@ -74,8 +77,53 @@ export default function createOrganizationApi ({
       .then(response => wrapOrganizationMembershipCollection(http, response.data), errorHandler)
   }
 
+  /**
+   * Gets an Organization Space Membership
+   * @memberof ContentfulOrganizationAPI
+   * @param {string} id - Organiztion Space Membership ID
+   * @return {Promise<OrganizationSpaceMembership.OrganizationSpaceMembership>} Promise for a Organization Space Membership
```
Comment (guilebarbosa) [inline lib/create-organization-api.js]: Can you change all docs to say Space Membership instead?
Also the return type declaration should be something like
```js
@return {Promise<SpaceMembership>} Promise for a Space Membership
```

File: lib/create-organization-api.js [hunk 10]
```diff
@@ -74,8 +77,53 @@ export default function createOrganizationApi ({
...
+      headers
+    })
+      .then(response => wrapOrganizationSpaceMembership(http, response.data, orgId), errorHandler)
+  }
+
+  /**
+   * Gets a collection of Organization Space Memberships
+   * @memberof ContentfulOrganizationAPI
+   * @param {object=} query - Object with search parameters. Check the <a href="https://www.contentful.com/developers/docs/javascript/tutorials/using-js-cda-sdk/#retrieving-entries-with-search-parameters">JS SDK tutorial</a> and the <a href="https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters">REST API reference</a> for more details.
+   * @return {Promise<OrganizationSpaceMembership.OrganizationSpaceMembershipCollection>} Promise for a collection of Organization Space Memberships
```
Comment (guilebarbosa) [inline lib/create-organization-api.js]: Same here for the docs. Could be something like:
```js
@return {Promise<SpaceMembership[]>} Promise for a Space Membership collection across all spaces in the organization
```

File: test/integration/organization-space-membership-integration.js (hunk 6 above)
Comment (guilebarbosa) [reply test/integration/organization-space-membership-integration.js]: Oh, sorry! 🤦🏻‍♂️
Yeah, that would be it.

File: lib/entities/organization-space-membership.js (hunk 8 above)
Comment (burakukula) [reply lib/entities/organization-space-membership.js]: 🧐hmmm. yeah.. I think so. i will test it and try to remove it.

**Comment by burakukula** [comment]
@guilebarbosa I implement changes that you suggested, it's got smaller and better 🎉thank you! If you could have a look again, when you have a moment, would be awesome. thx a lot!

**Comment by makinwab** [comment]
Please squash and merge using the sem ver commit pattern so that semantic release can be triggered appropriately. @burakukula

**Pull request merged**